XamarinCommunityToolkit/Views/CameraView/CameraView.shared.cs

```diff
@@ -14,27 +14,17 @@ public class CameraView : View
 
 		internal event EventHandler ShutterClicked;
 
-		public static readonly BindableProperty ShutterCommandProperty =
-			BindableProperty.Create(nameof(ShutterCommand),
-						typeof(ICommand),
-						typeof(CameraView));
+		internal static readonly BindablePropertyKey ShutterCommandPropertyKey =
+			BindableProperty.CreateReadOnly(nameof(ShutterCommand),
+				typeof(ICommand),
+				typeof(CameraView),
+				default,
+				BindingMode.OneWayToSource,
+				defaultValueCreator: ShutterCommandValueCreator);
 
-		public ICommand ShutterCommand
-		{
-			get => (ICommand)GetValue(ShutterCommandProperty);
-			set => SetValue(ShutterCommandProperty, value);
-		}
-
-		public static readonly BindableProperty ShutterCommandParameterProperty =
-			BindableProperty.Create(nameof(ShutterCommandParameter),
-						typeof(object),
-						typeof(CameraView));
+		public static readonly BindableProperty ShutterCommandProperty = ShutterCommandPropertyKey.BindableProperty;
 
-		public object ShutterCommandParameter
-		{
-			get => GetValue(ShutterCommandParameterProperty);
-			set => SetValue(ShutterCommandParameterProperty, value);
-		}
```

Comment:
What is the reasoning behind removing the `ShutterCommandParameter`?

I find parameters as implemented to be very useful when you need to pass a parameter to the ViewModel method that is being invoked by the `ICommand`. For example you can bind to non ViewModel objects such as a single item of a collection. 

I think having the `ShutterCommandParameter` will be useful when you need to pass additional information to the ViewModel.

A viewmodel example would look something like this
```c#
public class CameraViewModel
{
    public ICommand Shutter => new Command<object>(OnShutter);

    void OnShutter(object data)
    {
        // do something with the data 
    }
}
```

Reply:
The CommandParameter makes sense when you can implement your own Command, like `Button.Command`, this case `CameraView.ShutterCommand` is `readonly`, so the dev can't override it. If you see the code sample, I need to have a Command for the Button that triggers the `CameraView.ShutterCommand`. So, if you want to use some `CommandParameter` you should use the `Button.CommandParameter`. Is this makes sense for you?

Reply:
Thank you for the explanation, this does make sense. Since the `ICommand` is being instantiated by the toolkit a developer has no control on the method being invoked. This means the ViewModel just creates the `ICommand` property and it is automatically set.

There is a little bit of magic here, which is okay. I think we need to be diligent with our documentation on this. It will be very easy to get confused on how this property is instantiated. Especially if you set it in your ViewModel and it doesn't do anything.